src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -892,37 +935,54 @@ def is_frontend_port_present_in_host():
     return True
 
 
-def get_num_dpus():
+def get_dpu_info():
     """
-    Retrieves the number of DPUs from platform.json file.
-
-    Args:
+    Retrieves the DPU information from platform.json file.
 
     Returns:
-        A integer to indicate the number of DPUs.
+        A dictionary containing the DPU information.
     """
 
     platform = get_platform()
     if not platform:
-        return 0
+        return {}
 
-    # Get Platform path.
-    platform_path = get_path_to_platform_dir()
+    # Retrieve platform.json data
+    platform_data = get_platform_json_data()
+    if platform_data:
+        # Convert keys to lower case to avoid case sensitivity.
+        data = {k.lower(): v for k, v in platform_data.items()}
+        dpu_info = data.get('dpus', {})
```

Comment:
Can we make this function compatible with the DPU? E.g.:
```
if 'DPUS' in platform_data:
    return platform_data['DPUS']
elif 'DPU' in platform_data:
    return platform_data['DPU']
```

Reply:
Yes, good! I will modify it.

Reply:
@vvolam what is the intention of this API? Previously, you cared about DPUs which indicated this was supposed to be used by NPU ...but now it usage is on DPU as well?

Reply:
@prgeor Intention of this API is to return DPU information from the platform.json file. In case of NPU, it returns the information for all DPUs whereas in case of DPU, it returns only the DPU specific information.